setup/setup_all.sh

```diff
@@ -97,41 +97,44 @@ create_terraform_gcs_bucket() {
 # Run terraform to set up all GCP resources. (Setting up GKE by default)
 init_foundation() {
   # Init Terraform
-  cd "$BASE_DIR"/terraform/stages/foundation
+  cd "${BASE_DIR}"/terraform/stages/foundation
   terraform init -reconfigure -backend-config=bucket="${TF_BUCKET_NAME}"
   
   # Enabling GCP services first.
   terraform apply -target=module.project_services -target=module.service_accounts -auto-approve
-  
-  # Initializing Firebase (Only need this for the first time.)
-  # NOTE: the Firebase can only be initialized once (via App Engine).
-  terraform apply -target=module.firebase -var="firebase_init=true" -auto-approve
+
+  # Skip firebase init if running E2E tests as it is already done
+  if [[ "${RUN_E2E_TEST}" != "true" ]]; then
+    # Initializing Firebase (Only need this for the first time.)
+    # NOTE: the Firebase can only be initialized once (via App Engine).
+    terraform apply -target=module.firebase -var="firebase_init=true" -auto-approve
+  fi
```

Comment:
This is still not reliable, especially if anyone run setup_all.sh twice on their project. I'd suggest exploring with more on how gcloud CLI could find.

Reply:
I checked the document throughly there is no gcloud (or firebase) command to find out if firebase ha been initialized. The only other info gathered was that default app engine service account is created upon init and stays there unless someone explicitly deletes it.